pkg/storageincentives/redistribution/redistribution.go

```diff
@@ -89,6 +94,10 @@ func (c *contract) IsWinner(ctx context.Context) (isWinner bool, err error) {
 		return false, fmt.Errorf("IsWinner: results %v : %w", results, err)
 	}
 
+	if len(results) == 0 {
+		return false, errors.New("unexpected empty results")
+	}
+
 	return results[0].(bool), nil
```

Comment:
that's good idea, thankyou

Reply:
maybe we could have a function `checkResults` that does both length check and type cast check...

Reply:
one more note: 
it looks like `abi.ConvertType` method is used for similar usecase, but it can also panic if for same reason type from result does not match desired type

Reply:
thankyou, seems logical. Fixing those. 
@aloknerurkar we can merge this after releasing rc.